### DISCUSSION TITLE
Spawned random model falling through randomized terrain

### DISCUSSION THREAD

**Discussion by p-vega** [AndrejOrsula/drl_grasping#109] [Q&A]
Hi Andrej, 

I'm currently working on a grasp and lift project and I'm using this repository as a base (as I'm using a similar robot and tools). I have managed to use the Docker and also to do a local installation on my system but I'm having trouble with the Gazebo simulation. 
As described in #108 the randomized models spawned in the simulation sometimes fall through the randomized ground (sometimes even fall through the underworld collision plane). I've noticed that both the Docker and the local installation have the same behavior. So far I have tried to play with some friction parameters or even the physics rate but the problem persists. Do you have any idea about what could be done to improve this aspect of the simulation ? Any pointer would be greatly appreciated.

Thanks in advance !

**Comment by AndrejOrsula** [comment]
Hello,

That's a very good question! I tried checking it after it was mentioned in #108. I have not encountered it using Docker image for version `1.1.0`, but it occurs in `2.0.0`. I do not believe there is any significant difference in the way objects are spawned between these two versions, which could indicate that it has something to do with updated dependencies (Gazebo). Since it doesn't occur every time, it is quite difficult to debug, and I am not sure what is causing the issue. I tried playing with the physics parameters for a while, but I was not able to solve the issue from within the `2.0.0` container. Downgrading Gazebo libraries would probably be my next step to figure out if that's causing it, but I don't have ideas beyond that. As already mentioned in another discussion, upgrading Gazebo to `garden` would take a considerable amount of effort and I unfortunately don't have time for that due to other responsibilities.

**Comment by p-vega** [comment]
Yes, I've noticed that when "object_models_rollouts_num" and "terrain_model_rollouts_num" are set to 0 (so randomize_terrain and randomize_object_models are not called) this problem doesn't exist. So indeed the problem is related to scenario's insert_model and remove_model
